### ISSUE TITLE
TypeScript compilation: `declare const` is not de-referenced properly

### ISSUE THREAD

**Issue by iamakulov** [evanw/esbuild#227]
Hey,

I stumbled upon a ESBuild bug with the following piece of code (simplified from a real project):

```ts
export namespace Assets {
    export declare const service: string

    export function on(channel: string) {
        return someCall(service, channel) // ← A
    }
}

// Later: assign the actual `service` implementation
(Assets as any).service = getTheServiceFromSomewhere()
```

ESBuild seems to leave the `service` reference at line A as-is. This results in `Uncaught ReferenceError: service is not defined`.

#### TypeScript output ([playground](https://www.typescriptlang.org/play?ssl=1&ssc=1&pln=11&pc=55#code/KYDwDg9gTgLgBAOwIYFtgGcxIMbDgQXXWBnTgG8BYAKDjrlEljgBNhsAbJKPbCBdPGJQAbgEtcALjiCoYhAHMaNeg3DR4AMwCuCbDDH84-ABTYAFkgQJgHabPkKAlBRWr6PGNqgIZENADCSBwcJsLiuAA0cBZWNhxObnQAvjSp1MrUAPRZcAAySDDAUNJIRGIKvjDmeDhewXAABuESwI1wYihgHMBoCDCFhgg0JoTEpHBlkwgAnk4AdC24cAC8cAokACo1AMrFEcAAYlD+O-7AAO41PCZOQA)):

```js
export var Assets;
(function (Assets) {
    function on(channel) {
        // ↓ Note how `service` is de-referenced to `Assets.service`
        return someCall(Assets.service, channel);
    }
    Assets.on = on;
})(Assets || (Assets = {}));
// Later: assign the actual `service` implementation
Assets.service = getTheServiceFromSomewhere();
```

#### ESBuild output:

```js
// ...
var Assets;
(function(Assets2) {
    function on(channel) {
    // ↓ Note how `service` is kept as-is
    return someCall(service, channel);
    }
    Assets2.on = on;
})(Assets || (Assets = {}));
Assets.service = getTheServiceFromSomewhere();
```

Compiled with:

```sh
esbuild --bundle test.ts --outfile=out.js
```

**Comment by evanw** [comment]
Interesting. Thanks for reporting this. This will be good to fix.

**Comment by evanw** [comment]
This should be fixed in version 0.5.23.
